### ISSUE TITLE
templates: allow access to values of regex captures

### ISSUE THREAD

**Issue by adrian-gierakowski** [SpectoLabs/hoverfly#841]
**Is your feature request related to a problem? Please describe.**
I'd like to be able to transform data from request when templating the response

**Describe the solution you'd like**
in this particular case, simply being able to use data from regex capture (when using regex matcher) would be enough for me. Fox example, when using following matcher 
```json
"query": {
  "access_token": [
    {
      "matcher": "regex",
      "value": "(\\d*)_token"
    }
  ]
}
```

I'd like to be able to do something like: `{{ Request.RegexCaptures.QueryParam.access_token.[0] }}` to access the value captured by the `(\\d*)`

**Describe alternatives you've considered**
An alternative would be to provide a rich set of utility function for string processing etc. which could be used within the templates.

**Comment by tommysitu** [comment]
Hi @adrian-gierakowski, thanks for the suggestion. Providing some helper functions for string processing is good, as you can use it for other scenarios.

**Comment by tommysitu** [comment]
You should be able to do this using `{{ replace Request.QueryParam.access_token.[0] "_token" "" }}` in the upcoming release.

**Comment by tommysitu** [comment]
Fixed in v1.0.1
